subprojects/performance/src/performanceTest/groovy/org/gradle/performance/experiment/maven/JavaTestGradleVsMavenPerformanceTest.groovy

```diff
@@ -17,83 +17,70 @@
 package org.gradle.performance.experiment.maven
 
 import org.gradle.performance.AbstractGradleVsMavenPerformanceTest
-import org.gradle.performance.fixture.JavaSourceFileUpdater
+import org.gradle.performance.mutator.ApplyNonAbiChangeToJavaSourceFileMutator
 import spock.lang.Unroll
 
+import static org.gradle.performance.generator.JavaTestProject.MEDIUM_JAVA_MULTI_PROJECT
+import static org.gradle.performance.generator.JavaTestProject.MEDIUM_MONOLITHIC_JAVA_PROJECT
+
 /**
  * Performance tests aimed at comparing the performance of Gradle for compiling and executing test suites, making
  * sure we are always faster than Maven.
  */
 class JavaTestGradleVsMavenPerformanceTest extends AbstractGradleVsMavenPerformanceTest {
-    @Unroll("Gradle vs Maven #description build for #template")
-    def "cleanTest test performance test"() {
+
+    @Unroll
+    def "#gradleTasks on #testProject (Gradle vs Maven)"() {
         given:
         runner.testGroup = "Gradle vs Maven test build using Java plugin"
-        runner.testId = "$size $description with Java plugin"
-        equivalentGradleAndMavenBuilds(template, description, gradleTasks, equivalentMavenTasks)
-
-        when:
-        def results = runner.run()
+        runner.testProject = testProject
+        runner.jvmOpts = ["-Xms${testProject.daemonMemory}", "-Xmx${testProject.daemonMemory}"]
+        runner.args = testProject.parallel ? ['-T', testProject.parallelForks] : []
+        runner.gradleTasks = gradleTasks
+        runner.equivalentMavenTasks = equivalentMavenTasks
 
-        then:
-        results.assertComparesWithMaven(maxDiffMillis, maxDiffMB)
-
-        where:
-        template          | size     | description                 | gradleTasks           | equivalentMavenTasks | maxDiffMillis | maxDiffMB
-        'mediumWithJUnit' | 'medium' | 'runs tests only'           | ['cleanTest', 'test'] | ['test']             | 10000         | 100
-        'mediumWithJUnit' | 'medium' | 'clean build and run tests' | ['clean', 'test']     | ['clean', 'test']    | 5000          | 100
-    }
-
-    @Unroll("Gradle vs Maven #description build for #template")
-    def "build performance test"() {
-        given:
-        runner.testGroup = "Gradle vs Maven build using Java plugin"
-        runner.testId = "$size $description with Java plugin"
-        equivalentGradleAndMavenBuilds(template, description, gradleTasks, equivalentMavenTasks)
+        runner.configure()
```

Comment:
The individual tests shouldn't have to call this. Either we make it the hard convention that there is always one Gradle and one Maven invocation that is configured implicitly through a few fields or we just add a convenience method that takes some parameters. Having a mix of fields and a no-arg method call will be confusing for the next person looking at that base class.